wremnants/syst_tools.py

```diff
@@ -1,20 +1,24 @@
 import hist
 import numpy as np
 
-def scale_helicity_hist_to_variations(scale_hist, sum_helicity=False, sum_ptV=False, rebinPtV=0):
-
+def scale_helicity_hist_to_variations(scale_hist, sum_helicity=False, sum_ptV=False, hasNoHelicity=False, rebinPtV=0):
+    # hasNoHelicity=True allows for usage with QCD scale histograms made with no helicity splitting
+    
     s = hist.tag.Slicer()
     # select nominal QCD scales, but keep the sliced axis at size 1 for broadcasting
     nom_scale_hist = scale_hist[{"muRfact" : s[1.j:1.j+1], "muFfact" : s[1.j:1.j+1]}]
 
     # select nominal QCD scales and project down to nominal axes
-    nom_hist = nom_scale_hist[{"ptVgen" : s[::hist.sum], "chargeVgen" : s[::hist.sum], "helicity" : s[::hist.sum], "muRfact" : s[1.j], "muFfact" : s[1.j] }]
+    if hasNoHelicity:
+        nom_hist = nom_scale_hist[{"ptVgen" : s[::hist.sum], "chargeVgen" : s[::hist.sum], "muRfact" : s[1.j], "muFfact" : s[1.j] }]
+    else:
+        nom_hist = nom_scale_hist[{"ptVgen" : s[::hist.sum], "chargeVgen" : s[::hist.sum], "helicity" : s[::hist.sum], "muRfact" : s[1.j], "muFfact" : s[1.j] }]
 
     if rebinPtV > 0:
         scale_hist = scale_hist[{"ptVgen" : s[::hist.rebin(rebinPtV)]}]
         nom_scale_hist = nom_scale_hist[{"ptVgen" : s[::hist.rebin(rebinPtV)]}]
     
-    if sum_helicity:
+    if sum_helicity and not hasNoHelicity:
```

Comment:
Just check if it has a helicity axis?

Reply:
Done, the whole function is modified accordingly